src/lib.rs

```diff
@@ -62,41 +65,88 @@ pub fn run(args: Args) {
             .expect(&format!("Unknown capture name: `{}`", capture_name))
     });
 
-    let (sender, receiver) = mpsc::channel::<DirEntry>();
-    let handle = thread::spawn(move || {
-        let thread_pool = ThreadPool::new(num_cpus::get());
-        while let Ok(project_file_dir_entry) = receiver.recv() {
-            thread_pool.execute({
-                let query = query.clone();
-                let args_filter = args.filter.clone();
-                let args_filter_arg = args.filter_arg.clone();
-                move || {
-                    let mut printer = grep::printer::Standard::new_no_color(io::stdout());
-                    let path = project_file_dir_entry.path();
-
-                    let matcher = TreeSitterMatcher::new(
-                        &query,
-                        capture_index,
-                        language,
-                        args_filter,
-                        args_filter_arg,
-                    );
-
-                    SearcherBuilder::new()
-                        .multi_line(true)
-                        .build()
-                        .search_path(&matcher, path, printer.sink_with_path(&matcher, path))
-                        .unwrap();
-                }
+    get_project_file_walker(&*supported_language)
+        .into_parallel_iterator()
+        .for_each({
+            |project_file_dir_entry| {
+                let mut printer = grep::printer::Standard::new_no_color(io::stdout());
+                let path = project_file_dir_entry.path();
+
+                let matcher = TreeSitterMatcher::new(
+                    &query,
+                    capture_index,
+                    language,
+                    args.filter.clone(),
+                    args.filter_arg.clone(),
+                );
+
+                SearcherBuilder::new()
+                    .multi_line(true)
+                    .build()
+                    .search_path(&matcher, path, printer.sink_with_path(&matcher, path))
+                    .unwrap();
+            }
+        });
+}
+
+trait IntoParallelIterator {
```

Comment:
Expressing this via a trait was wholly unnecessary (I [initially](https://github.com/helixbass/tree-sitter-grep/pull/11/commits/a5c4cbc7bfdb1f85046ad848b882135a99d3d0e7) had it as
```
WalkParallelIterator::new(get_project_file_walker(&*supported_language))
    .par_bridge()
    .for_each(...)
```
), I just thought the `.into_parallel_iterator().for_each(...)` chaining (+ encapsulating of `.par_bridge()`) was stylish